src/system/OS_info/os_info_windows.cpp

```diff
@@ -80,10 +81,58 @@ static std::string version_name() {
 	return ret.substr(0, ret.find('|'));
 }
 
+unsigned int build_number() {
+	HKEY hkey{};
+	if(RegOpenKeyExA(HKEY_LOCAL_MACHINE, R"(Software\Microsoft\Windows NT\CurrentVersion)", 0, KEY_READ, &hkey))
+		return {};
+
+	DWORD ubr      = 0;
+	DWORD ubr_size = sizeof(ubr);
+	if(!RegQueryValueExA(hkey, "UBR", nullptr, nullptr, reinterpret_cast<LPBYTE>(&ubr), &ubr_size))
+		return ubr;
+
+	// Fall back to BuildLabEx in the early version of Windows 8.1 and less.
+	constexpr auto buildlabex_name = "BuildLabEx";
+	DWORD buildlabex_size          = 0;
+	if(RegQueryValueExA(hkey, buildlabex_name, nullptr, nullptr, nullptr, &buildlabex_size))
+		return {};
+
+	std::vector<char> buildlabex_buffer(buildlabex_size);
+	if(RegQueryValueExA(hkey, buildlabex_name, nullptr, nullptr, reinterpret_cast<LPBYTE>(buildlabex_buffer.data()), &buildlabex_size))
+		return {};
+
+	const std::string buildlabex{buildlabex_buffer.begin(), buildlabex_buffer.end()};
```

Comment:
why the copy here? RQVEA directly into a std::string

Reply:
First I tried to get registry value directly into a std::string by passing `buildlabex.data()`, but I realized that it returns `const char*` in C++14. I will do it again by passing `&buildlabex[0]`.